server/crawler/crawler.go

```diff
@@ -149,30 +149,70 @@ func convertRepoContentToApi(kurtosisPackageContent *KurtosisPackageContent) *ge
 	var kurtosisPackageArgsApi []*generated.PackageArg
 	for _, arg := range kurtosisPackageContent.PackageArguments {
 		var convertedPackageArg *generated.PackageArg
-		convertedPackageArgType, ok := generated.PackageArgType_value[strings.ToUpper(arg.Type)]
-		if ok {
-			convertedPackageArg = api_constructors.NewPackageArg(
-				arg.Name,
-				arg.IsRequired,
-				generated.PackageArgType(convertedPackageArgType))
-		} else {
-			if arg.Type != "" {
-				logrus.Warnf("Argument type '%s' for argument '%s' in package '%s' is no known and will be dropped",
+		convertedPackageArgTypeV2Ptr, ok := convertArgumentType(arg.Type)
+		if !ok {
+			if arg.Type != nil {
+				logrus.Warnf("Argument type '%s' for argument '%s' in package '%s' could not be parsed to the new format",
 					arg.Type, arg.Name, kurtosisPackageContent.Identifier)
 			}
-			convertedPackageArg = api_constructors.NewUntypedPackageArg(arg.Name, arg.IsRequired)
 		}
+
+		var convertedPackageArgTypeV1Ptr *generated.ArgumentValueType
+		if arg.Type != nil {
+			convertedPackageArgTypeV1Raw, ok := generated.ArgumentValueType_value[strings.ToUpper(arg.Type.Type.String())]
```

Comment:
Could `arg.Type.Type` be nil? (currently only checking the first `Type`)

Reply:
It cannot, it is a required field. I'll add it as a comment here